First version of "Updating Drupal" docs

ysuarez opened the pull request:
**GitHub Issue**: (https://github.com/Islandora/documentation/issues/1040)

* Other Relevant Links (Google Groups discussion, related pull requests, Release pull requests, etc.)

# What does this Pull Request do?

Adds new documentation for updating Drupal core and modules.

# How should this be tested?

Review what was written and suggest changes or corrections.

# Interested parties
@Islandora/8-x-committers
@dannylamb

ysuarez committed:
First version of "Updating Drupal" docs

ysuarez commented:
I still need to write out instructions for updating individual modules, but it is a start.

manez commented on docs/technical-documentation/updating_drupal.md:
```diff
@@ -0,0 +1,67 @@
+
+"[Composer](https://getcomposer.org/) is a [dependency manager](https://en.wikipedia.org/wiki/Package_manager) for PHP. Drupal core uses Composer to manage core dependencies like Symfony components and Guzzle."
+
+Taken from: https://www.drupal.org/docs/develop/using-composer/using-composer-with-drupal
```
Would suggest making this `"[Source](https://www.drupal.org/docs/develop/using-composer/using-composer-with-drupal)"` instead.

manez commented on docs/technical-documentation/updating_drupal.md:
```diff
@@ -0,0 +1,67 @@
+## Updating Drupal Core
+Over time new versions of Drupal “core” are released, and Islandora users are encouraged to install official Drupal core updates and security patches. On the other hand “alpha” and “beta" versions of Drupal core should only be installed if by advanced users for testing purposes. 
+
+The Islandora community STRONGLY recommends that the "Composer" method of upgrading Drupal core be used with Islandora 8 as mentioned here:
```
`"[here](https://www.drupal.org/docs/8/update/update-core-via-composer)"`

manez commented on docs/technical-documentation/updating_drupal.md:
```diff
@@ -0,0 +1,67 @@
+
+### Here is an overview of the steps for updating Drupal core using Composer
+
+NOTE: First make sure you have made database and file back ups.
```
If you do this as`"!!! note "Back Up" First make sure you have made database and file back ups."`, it will parse it as a highlighted note in the final docs, like on [this page](https://islandora.github.io/documentation/user-documentation/content_types/)

manez commented on docs/technical-documentation/updating_drupal.md:
```diff
@@ -0,0 +1,67 @@
+
+If there is no line starting with drupal/core, Composer isn't aware of any update. If there is an update, continue with the commands below.
+
+If there is no line starting with drupal/core, Composer isn't aware of any update. If there is an update, continue with the commands below.
```
Duplicate line

manez commented on docs/technical-documentation/updating_drupal.md:
```diff
@@ -0,0 +1,67 @@
+
+If you want to know all packages that will be updated by the update command, use the --dry-run option first.
+
+NOTE: Islandora 8 is configured to use a fork of drupal-composer/drupal-project which requires a specific composer syntax used above compared to other Drupal 8 sites. In addition, if you are upgrading from 8.5 to 8.7, you need to replace "~8.5.x" with "^8.7.0" for drupal/core and webflo/drupal-core-require-dev in composer.json. 
```
See above- there's a format to make notes "pop" 😄

manez commented on docs/technical-documentation/updating_drupal.md:
```diff
@@ -0,0 +1,67 @@
+
+NOTE: Islandora 8 is configured to use a fork of drupal-composer/drupal-project which requires a specific composer syntax used above compared to other Drupal 8 sites. In addition, if you are upgrading from 8.5 to 8.7, you need to replace "~8.5.x" with "^8.7.0" for drupal/core and webflo/drupal-core-require-dev in composer.json. 
+
+Taken from:
```
`[source](https://www.drupal.org/docs/8/update/update-core-via-composer#s-one-step-update-instruction)`

manez commented on docs/technical-documentation/updating_drupal.md:
```diff
@@ -0,0 +1,67 @@
+
+`drush updatedb`
+
+4) Clear the cache using drush cache:rebuild, or use the web admin user interface.
```
putting `` marks around snippets can make them a little easier to pick out from the text, like: ` drush cache:rebuild`

manez commented on docs/technical-documentation/updating_drupal.md:
```diff
@@ -0,0 +1,67 @@
+
+## Updating Drupal Modules
+
+Islandora 8 uses several general Drupal modules and some specialized Islandora Drupal modules, and over time new versions of these modules are released. There are two approaches to updating Drupal modules in Islandora using Composer or updating modules individually. Islandora 8 uses Composer to determine which Drupal module versions should be installed for each release of Islandora 8. Therefore i ...
```
Add a colon after "There are two approaches to updating Drupal modules in Islandora" ?

manez commented on docs/technical-documentation/updating_drupal.md:
```diff
@@ -0,0 +1,67 @@
+Taken from: https://www.drupal.org/docs/develop/using-composer/using-composer-with-drupal
+
+## Updating Drupal Core
+Over time new versions of Drupal “core” are released, and Islandora users are encouraged to install official Drupal core updates and security patches. On the other hand “alpha” and “beta" versions of Drupal core should only be installed if by advanced users for testing purposes. 
```
"should only be installed if by advanced users" drop the 'if'here, I think.

manez reviewed (changes requested):
A few minor suggestions, but overall a really clear and helpful piece of documentation. Nice work!

ysuarez committed:
Made changes to content based on feedback

ysuarez commented:
Thanks @manez! Let me you think of the changes I made based on your feedback.

manez reviewed (approved):
Looks good!

dannylamb reviewed (approved):
Looks good.  You want to add in an entry for it in the `mkdocs` file so it shows up in the TOC?  Then I'll merge.

ysuarez commented:
@manez @dannylamb 

Should I update mkdocs.yml so this content shows up in the “Administrator Documentation” section (that only contains “RDF Generation”), or should it be listed in the “Developer Documentation" section? (Of course this depends how folks interpret developer v.s. administrator documentation.)

I also wonder if we would want to put the "Installing modules" content that currently lives in the “Developer Documentation", also under the “Administrator Documentation”?

manez commented:
It seems to me that both are a better fit in the Administrator documentation, but I'll defer to @dannylamb  on the Grand Plan. https://islandora.github.io/documentation/installation/ probably eblongs there too, but its off in its own place because of significance

dannylamb commented:
I think @rosiel made that section as more of a "Repository Administrator" than a "Systems Administrator".  We should probably sort that out.  Common sysadmin tasks need a good clear spot, preferably near the installation instructions.

rosiel commented:
Yeah, that's how it was meant. Sweet docs!

I always wonder, when I land on one of our docs, how much of it is "Drupal" and whether any is "Islandora specific". In this case...

This page describes how to update Drupal and its modules using Composer. If you installed Islandora using the Islandora Playbook, then your Drupal was installed by Composer, so it is considered best practice to continue using Composer for updates. The method on this page is not specific to Islandora, and does not (yet) include how to update Islandora Features.

Would you consider surfacing some of that (which you explain well throughout the document) right at the top?

ysuarez committed:
Added introduction text from Rosie

ysuarez commented:
@rosiel Thanks for the feedback. I don't know what you mean by "surfacing," for now I just added your suggested text to the top. What other edits should I make?

Should I create another Github issue to explore creating new sections like "sysadmin documentation," and perhaps rename "administration documentation" to "repository administration documentation"?

dannylamb commented:
I would do just that @ysuarez.  Make a sysadmin section and add the word "repository" to the current admin section.

ysuarez committed:
Update to documentation menu organization, new introduction to section

ysuarez commented:
@dannylamb @rosiel @manez 

Here is what I have done so far...
1) I added Rosie's intro text as is (it looked great to me)

2) I updated the documentation menu organization
* I renamed "Administrator Documentation" to "Repository Administrator Documentation"
* I added a new "System Administrator Documentation"
I added my new section, updating Drupal, here.
I also moved Installing Modules here too.

Thoughts?

dannylamb reviewed (approved):
Looks good to me @ysuarez.  Much appreciated, as always.
